Libraries/src/Amazon.Lambda.Annotations/SQS/SQSEventAttribute.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+
+namespace Amazon.Lambda.Annotations.SQS
+{
+    /// <summary>
+    /// This attribute defines the SQS event source configuration for a Lambda function.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+    public class SQSEventAttribute : Attribute
```

Comment:
Except for `Queue` all other properties are optional. 

.NET attributes cannot be nullable. To work around this, I added nullable backing fields to all optional properties and added an internal `Is<PropertyName>Set` method to identify which properties were explicitly set the customer.
These internal methods are used by the `CloudFormationWriter` while deciding which properties to write in the CF template.

Reply:
Please add this context to the code itself.